consensus/istanbul/backend/snapshot.go

```diff
@@ -220,22 +219,13 @@ func (s *Snapshot) apply(headers []*types.Header, gov governance.Engine, addr co
 			}
 		}
 	}
-	snap.Number += uint64(len(headers))
-	snap.Hash = headers[len(headers)-1].Hash()
-
-	if snap.ValSet.Policy() == istanbul.WeightedRandom {
-		snap.ValSet.SetBlockNum(snap.Number)
-
-		bigNum := new(big.Int).SetUint64(snap.Number)
-		if chain.Config().IsRandaoForkBlockParent(bigNum) {
-			// The ForkBlock must select proposers using MixHash but (ForkBlock - 1) has no MixHash. Using ZeroMixHash instead.
-			snap.ValSet.SetMixHash(params.ZeroMixHash)
```

Comment:
@ian0371 @blukat29  This is the only place I think the semantics have changed after this PR. Please carefully review this 
- In other code, MixHash values are always considered as `nil` before Randao, but it is assigning value here 
- I added `zeroMixHash` assigning logic for RandaoParent block to replace this code.

Reply:
I think the `snapshot.mixHash` of RandaoParent must have been set; otherwise, proposer selection for Randao block will fail:
https://github.com/klaytn/klaytn/blob/8223d5bd6af76133d7fb6da4a3d4b575863fc173/consensus/istanbul/validator/weighted.go#L279-L282

With the new logic, it seems that the `snapshot.mixHash` of the RandaoParent is nil.

Reply:
The unit test breaks with this change.

1. In `tests/randao_fork_test.go:L32`, set `numNodes = 4`
2. Run the test
	```
	go test -v -run=TestRandao_Deploy
	```
3. Panics in ValSet
	```
	    randao_fork_test.go:52: block[ 13] txs=0 mixHash=
	panic: runtime error: slice bounds out of range [:8] with capacity 0
	
	goroutine 557 [running]:
	github.com/klaytn/klaytn/consensus/istanbul/validator.SelectRandaoCommittee({0x1404397e5c0?, 0x104ab3864?, 0x10?}, 0xf?, {0x1070a7718?, 0x104b1567c?, 0x1402eda1ae8?})
	        /Users/krust/klay/klaytn/consensus/istanbul/validator/validator.go:156 +0x134
	github.com/klaytn/klaytn/consensus/istanbul/validator.weightedRandomProposer({0x10633e1d0?, 0x14042277080}, {0x70, 0x99, 0x79, 0x70, 0xc5, 0x18, 0x12, 0xdc, ...}, ...)
	        /Users/krust/klay/klaytn/consensus/istanbul/validator/weighted.go:285 +0xb8
	github.com/klaytn/klaytn/consensus/istanbul/validator.(*weightedCouncil).CalcProposer(0x14042277080, {0x70, 0x99, 0x79, 0x70, 0xc5, 0x18, 0x12, 0xdc, 0x3a, ...}, ...)
	        /Users/krust/klay/klaytn/consensus/istanbul/validator/weighted.go:524 +0xd8
	github.com/klaytn/klaytn/consensus/istanbul/core.(*core).startNewRound(0x140000f9b80, 0x140000a2220)
	        /Users/krust/klay/klaytn/consensus/istanbul/core/core.go:280 +0xe5c
	github.com/klaytn/klaytn/consensus/istanbul/core.(*core).handleFinalCommitted(0x140000f9b80)
	        /Users/krust/klay/klaytn/consensus/istanbul/core/final_committed.go:30 +0xe8
	github.com/klaytn/klaytn/consensus/istanbul/core.(*core).handleEvents(0x140000f9b80)
	        /Users/krust/klay/klaytn/consensus/istanbul/core/handler.go:141 +0x114
	created by github.com/klaytn/klaytn/consensus/istanbul/core.(*core).Start
	        /Users/krust/klay/klaytn/consensus/istanbul/core/handler.go:36 +0x70
	exit status 2
	FAIL    github.com/klaytn/klaytn/tests  17.118s
	```